client/src/components/NavBar/NavBar.jsx

```diff
@@ -0,0 +1,71 @@
+import React, { useState, useEffect } from "react";
+import './NavBar.css';
+
+const NavBar = ({ title }) => {
+  const [windowWidth, setWindowWidth] = useState(window.innerWidth);
+  const [showMenu, setShowMenu] = useState(false); // State for hamburger menu visibility
+  const [showAbout, setShowAbout] = useState(false); // State for showing the About component
+
+  // Update windowWidth state on window resize
+  useEffect(() => {
+    const handleResize = () => setWindowWidth(window.innerWidth);
+    window.addEventListener("resize", handleResize);
```

Comment:
I dont know much about this window resize event listener so im going to assume its alright as i saw it working before